frontend/model/contracts/group.js

```diff
@@ -141,6 +142,47 @@ function isActionYoungerThanUser (actionMeta: Object, userProfile: ?Object): boo
   return compareISOTimestamps(actionMeta.createdDate, userProfile.joinedDate) > 0
 }
 
+function updateGroupStreaks ({ state, getters }) {
+  const streaks = state.streaks
+  const cPeriod = getters.groupSettings.distributionDate
+  const thisPeriodPayments = getters.paymentsForPeriod(cPeriod)
+  const thisPeriodDistribution = adjustedDistribution({
+    distribution: unadjustedDistribution({
+      haveNeeds: getters.haveNeedsForThisPeriod(cPeriod),
+      minimize: getters.groupSettings.minimizeDistribution
+    }) || [],
+    payments: getters.paymentsForPeriod(cPeriod),
+    dueOn: getters.dueDateForPeriod(cPeriod)
+  })
+
+  // --- update 'fullMonthlyPledgesCount' streak ---
+  // if the group has made 100% pledges in this period, +1 the streak value.
+  // or else, reset the value to '0'
+  Vue.set(
+    streaks,
+    'fullMonthlyPledges',
+    thisPeriodDistribution.length === 0 ? streaks.fullMonthlyPledges + 1 : 0
+  )
+
+  // --- update 'onTimePayments' streaks for 'pledging' members of the group ---
+  const filterMyItems = (array, username) => array.filter(item => item.from === username)
+  for (const username in getters.groupProfiles) {
+    if (getters.groupProfiles[username].incomeDetailsType !== 'pledgeAmount') continue
```

Comment:
Because users can change this value during a payment period after it has already been locked, we shouldn't rely on / use this value.

Instead, use the `haveNeedsSnapshot` for the current payment period to figure out who was marked as being a pledger.